src/Microsoft.Security.Utilities.Core/IdentifiableScan.cs

```diff
@@ -249,6 +286,14 @@ public IEnumerable<Detection> DetectSecrets(string input)
         {
             yield return detection;
         }
+
+        if (this.backupSecretMasker != null)
```

Comment:
>backupSecretMasker [](http://example.com/codeflow?start=17&length=18)

currently, we only apply the back-up masker when scanning textual data, received as a populated string.

to make everything stream friendly would require getting into the IRegexPattern interface and some other places.